worker/peergrouper/controllertracker.go

```diff
@@ -87,11 +87,10 @@ func (c *controllerTracker) Addresses() network.SpaceAddresses {
 
 // SelectMongoAddress returns the best address on the controller node for MongoDB peer
 // use, using the input space.
-// An error is returned if the empty space is supplied.
 func (c *controllerTracker) SelectMongoAddressFromSpace(port int, space network.SpaceInfo) (string, error) {
-	if space.Name == network.DefaultSpaceName {
+	if space.ID == network.DefaultSpaceId {
```

Comment:
We know that this path will not be taken based on the choice of method to select a Mongo address.

I would add a TODO regarding removing this guard, or in fact just remove it now in anticipation of always using filtering by space.

Reply:
added a todo